src/Pages/Account/Account.js

```diff
@@ -0,0 +1,421 @@
+import React from "react";
+import { SERVER, SIGNUP_API, SIGNIN_API } from "../../config";
+import { Link } from "react-router-dom";
+import "./Account.scss";
+
+class Account extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      firstname: "",
+      firstnamehasValue: true,
+      lastname: "",
+      lastnamehasValue: true,
+      email: "",
+      password: "",
+      passwordhasValue: true,
+      checkPw: "",
+      checkPwhasValue: true,
+      adagreement: false,
+      serviceagreement: false,
+      serviceagreementhasValue: true,
+    };
+  }
+
+  // 인풋의 변화연동
+  firstnameInput = (e) => {
+    const firstnameValue = e.target.value;
+    this.setState(
+      {
+        firstname: firstnameValue,
+      },
+      () =>
+        this.setState({
+          firstnamehasValue: this.state.firstname.length > 0 ? true : false,
+        }),
+    );
+  };
+
+  lastnameInput = (e) => {
+    const lastnameValue = e.target.value;
+    this.setState({ lastname: lastnameValue }, () =>
+      this.setState({
+        lastnamehasValue: this.state.lastname.length > 0 ? true : false,
+      }),
+    );
+  };
+
+  emailInput = (e) => {
+    const emailValue = e.target.value;
+    this.setState({ email: emailValue }, () =>
+      this.setState({
+        emailhasValue: this.state.email.length > 0 ? true : false,
+      }),
+    );
+  };
+
+  passwordInput = (e) => {
+    const passwordValue = e.target.value;
+    this.setState({ password: passwordValue }, () =>
+      this.setState({
+        passwordhasValue: this.state.password.length > 0 ? true : false,
+      }),
+    );
+  };
+
+  checkPwInput = (e) => {
+    const checkPwValue = e.target.value;
+    this.setState({ checkPw: checkPwValue }, () =>
+      this.setState({
+        checkPwhasValue: this.state.checkPw.length > 0 ? true : false,
+      }),
+    );
+  };
+
+  adagreementInput = (e) => {
+    const adagreementValue = e.target.checked;
+    this.setState({ adagreement: adagreementValue });
+  };
```

Comment:
```suggestion
  adagreementInput = (e) => {
    const { checked } = e.target;
    this.setState({ adagreement: checked });
  };
```
- 객체 구조 분해 할당을 바로 적용할 수 있습니다.

Reply:
준식님 앞서 주신 댓글에서

```
handleInput = (e) => {
    const { name, value, checked } = e.target;

    this.setState(
      {
        [name]: value,
        [name]: checked,
      },
}
```

와 같이 checkbox를 합쳐 축약할 수는 없는 건가요?

Reply:
수정했습니다 ;)